packages/app/src/pages/Specs/Runner.vue

```diff
@@ -51,6 +51,18 @@ const query = useQuery({
   pause: isRunMode && window.top === window,
 })
 
+const queryInterval = setInterval(() => {
+  // it's possible that this route will render before the cache is populated, especially from tests
+  // that visit this route directly, so we can poll and re-execute this query until the cache has
```

Comment:
Are you sure about this? How did we determined this? This should be impossible - we pre-hydrate the GraphQL client with server-rendered data:

https://github.com/cypress-io/cypress/blob/10.0-release/packages/data-context/src/sources/HtmlDataSource.ts#L89-L121

Specifically this line: https://github.com/cypress-io/cypress/blob/844c6cefd675ecd3fe59f91f3d612065eed3a385/packages/data-context/src/sources/HtmlDataSource.ts#L112

We then hydrate here when creating the urql client https://github.com/cypress-io/cypress/blob/10.0-release/packages/frontend-shared/src/graphql/urqlClient.ts#L122-L124

Which is called **before** we mount the app in an synchronous fashion: https://github.com/cypress-io/cypress/blob/10.0-release/packages/app/src/main.ts#L39. There should be no possibility of rendering anything when the GraphQL client has not got data.

@marktnoonan I could be missing something. I also don't fully understand the flake or how it manifests, maybe we cna pair and you can walk me through what the fix you've implemented does and how to reproduce the flake (if we can reproduce it).

Reply:
Yeah totally, let's pair 👍 . For more context, the flake manifests like this in videos from failing runs in CI:

<img width="1308" alt="Screen Shot 2022-03-11 at 8 59 53 AM" src="https://user-images.githubusercontent.com/8340719/158145784-c28cb5a1-8b50-4541-a635-8720fbe73fcb.png">
 
 We visit the runner page but the runner doesn't render, due to the conditional here being false:
 
 ```html
    <SpecRunnerContainerOpenMode
      v-else-if="query.data.value?.currentProject?.specs"
      :gql="query.data.value"
    />
 ```
  
Since we are `cache-only` on that query, if we load the page and the value in the cache is not there, we get stuck in this state, so my first fix was to flip to `cache-first` - this fixes the flake and would have been my PR but reintroduces an older bug when new specs are created. I suspect that some of the work around actually populating the cache in urql's gotta be async for us to be seeing this - if we can find that and wait for it, that would be great, but if not, polling will at least let tests visit this page directly without getting stuck.